File: java/0448_findDisappearedNumbers.java
class Solution {
    public List<Integer> findDisappearedNumbers(int[] nums) {
        for(var i=0;i<nums.length;i++)
            if(nums[Math.abs(nums[i])-1]>0)
                nums[Math.abs(nums[i])-1] *= -1;
        List<Integer> res = new ArrayList<>();
        for(var i=0;i<nums.length;i++)
            if(nums[i]>0)
                res.add(i+1);
        return res;
    }
}

File: cpp/0448_findDisappearedNumbers.cpp
// 1.类似0041，遍历2遍数组：
// （1）先标记在目标范围内的数加负号（如果已经标记，则不加）
// （2）再遍历将未标记的数加入返回的结果数组
class Solution {
public:
    vector<int> findDisappearedNumbers(vector<int>& nums) {
        for(int i=0;i<nums.size();i++)
        {
            if(nums[abs(nums[i])-1]>0)
                nums[abs(nums[i])-1] *= -1;
        }
        vector<int> res;
        for(int i=0;i<nums.size();i++)
            if(nums[i]>0)
                res.push_back(i+1);
        return res;
    }
};
